great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -295,27 +343,250 @@ def run(
+        for (
+            expectation_configuration_builder_config
+        ) in expectation_configuration_builder_configs:
+            expectation_configuration_builder_name: str = (
+                expectation_configuration_builder_config["expectation_type"]
+            )
+            if (
+                expectation_configuration_builder_name
+                in current_expectation_configuration_builders
+            ):
+                expectation_configuration_builder: ExpectationConfigurationBuilder = (
+                    current_expectation_configuration_builders[
+                        expectation_configuration_builder_name
+                    ]
+                )
+                effective_expectation_configuration_builder_configs.append(
+                    RuleBasedProfiler._reconcile_expectation_configuration_builder(
+                        expectation_configuration_builder=expectation_configuration_builder,
+                        expectation_configuration_builder_config=expectation_configuration_builder_config,
+                    )
+                )
+            else:
+                effective_expectation_configuration_builder_configs.append(
+                    expectation_configuration_builder_config
+                )
+
+        return effective_expectation_configuration_builder_configs
+
+    @staticmethod
+    def _reconcile_expectation_configuration_builder(
+        expectation_configuration_builder: ExpectationConfigurationBuilder,
+        expectation_configuration_builder_config: dict,
+    ) -> dict:
+        effective_expectation_configuration_builder_config: dict = {}
+
+        key: str
+        value: Any
+        current_value: Any
+        for key, value in expectation_configuration_builder_config:
+            if hasattr(expectation_configuration_builder, f"{key}"):
+                current_value = getattr(expectation_configuration_builder, key)
+                effective_expectation_configuration_builder_config[key] = (
+                    value or current_value
+                )
+            else:
+                effective_expectation_configuration_builder_config[key] = value
+
+        return effective_expectation_configuration_builder_config
+
+    def _get_rules_as_dict(self) -> Dict[str, Rule]:
+        rule: Rule
+        return {rule.name: rule for rule in self._rules}
+
+    @property
+    def variables(self) -> Optional[ParameterContainer]:
+        # Returning a copy of the "self._variables" state variable in order to prevent write-before-read hazard.
+        return copy.deepcopy(self._variables)
+
+    @variables.setter
+    def variables(self, value: Optional[ParameterContainer]):
+        self._variables = value
+
+    @property
+    def rules(self) -> List[Rule]:
+        return list(self._get_rules_as_dict().values())
 
-        return report_object
+    @rules.setter
+    def rules(self, value: List[Rule]):
+        self._rules = value
```

Comment:
Same note about annotation.

Reply:
@cdkini Do we ever need to annotate the return of a setter?

Reply:
Same note as above.